src/ast.c

```diff
@@ -1037,6 +1037,15 @@ JL_DLLEXPORT int jl_operator_precedence(char *sym)
     return res;
 }
 
+JL_DLLEXPORT int jl_is_dotop_named(char *sym)
+{
+    jl_ast_context_t *ctx = jl_ast_ctx_enter();
+    fl_context_t *fl_ctx = &ctx->fl;
+    int res = fl_applyn(fl_ctx, 1, symbol_value(symbol(fl_ctx, "dotop-named?")), symbol(fl_ctx, sym));
+    jl_ast_ctx_leave(ctx);
+    return res;
+}
```

Comment:
Can't we just check if the first character is a dot?   I'm a little concerned at the interpreter overhead of calling back to scheme for every symbol.

Reply:
I guess if we only call it for top-level symbol expressions it isn't a problem.

Reply:
Yes, only checking the first symbol for a dot in the c code makes sense. All the magic then happens in the scheme code afterwards anyways.